src/v/cluster/partition_manager.cc

```diff
@@ -354,12 +370,53 @@ ss::future<> partition_manager::shutdown(const model::ntp& ntp) {
           "manager",
           ntp)));
     }
-
+    auto shutdown_state = start_tracking_shutdown(partition);
     // remove partition from ntp & raft tables
     _ntp_table.erase(ntp);
     _raft_table.erase(partition->group());
 
-    return do_shutdown(partition);
+    return do_shutdown(std::move(shutdown_state));
+}
+
+partition_manager::partition_shutdown_state::partition_shutdown_state(
+  ss::lw_shared_ptr<cluster::partition> p)
+  : partition(std::move(p))
+  , stage(partition_manager::partition_shutdown_stage::shutdown_requested)
+  , last_update_timestamp(ss::lowres_clock::now()) {}
+
+void partition_manager::partition_shutdown_state::update(
+  partition_shutdown_stage s) {
+    stage = s;
+    last_update_timestamp = ss::lowres_clock::now();
+}
+
+partition_manager::partition_shutdown_state
+partition_manager::start_tracking_shutdown(
+  ss::lw_shared_ptr<cluster::partition> partition) {
+    partition_shutdown_state state(std::move(partition));
+    _partitions_shutting_down.push_back(state);
+    return state;
+}
+
+void partition_manager::check_partitions_shutdown_state() {
+    const auto now = ss::lowres_clock::now();
+    for (auto& state : _partitions_shutting_down) {
+        if (state.last_update_timestamp < now - _partition_shutdown_timeout()) {
+            clusterlog.error(
+              "partition {} shutdown timedout, current shutdown stage: {} time "
```

Comment:
nit: "timedout" may suggest that we are abandoning it, even though we are not, maybe another word will be better